### ISSUE TITLE
malloc: Heap corruption detected and segfault when running demo example

### ISSUE THREAD

**Issue by sahuguet** [ywelsch/duckdb-prql#10]
```
duckdb(24577,0x16ffd3000) malloc: Heap corruption detected, free list is damaged at 0x600003aacec0
*** Incorrect guard value: 78611779014145
duckdb(24577,0x16ffd3000) malloc: *** set a breakpoint in malloc_error_break to debug
```
and
```
zsh: segmentation fault  duckdb --unsigned
```

when running the demo example.

version = v0.9.2 3c695d7ba9
OS = Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000 arm64

**Comment by ywelsch** [comment]
Thanks for reporting this. I will take a look!

**Comment by dannypage** [comment]
Got the same thing today on v0.10 and Mac 14.3.1 (23D60) - M1 Max 64gb ram

```
duckdb(4508,0x1e1e7dc40) malloc: Heap corruption detected, free list is damaged at 0x6000023f6480
*** Incorrect guard value: 78621966296065
duckdb(4508,0x1e1e7dc40) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      duckdb -unsigned
```

Tried in an in-memory database & in a file database.

**Comment by ywelsch** [comment]
This is now fixed in https://github.com/ywelsch/duckdb-prql/commit/a1b156ddcdb8cc3d1abf53622e3c3d214c136d11, so in 30mins from now the binaries should be available for duckdb 0.10.0 (I'm only doing releases for the latest duckdb).
